scipy/cluster/tests/test_hierarchy.py

```diff
@@ -322,79 +322,89 @@ def test_leaders_single(self, xp):
 
 class TestIsIsomorphic:
 
-    @skip_if_array_api
-    def test_is_isomorphic_1(self):
+    @skip_if_xp_not_numpy(reason='`is_isomorphic` only supports NumPy backend')
+    @array_api_compatible
+    def test_is_isomorphic_1(self, xp):
         # Tests is_isomorphic on test case #1 (one flat cluster, different labellings)
-        a = [1, 1, 1]
-        b = [2, 2, 2]
-        assert_(is_isomorphic(a, b))
-        assert_(is_isomorphic(b, a))
+        a = xp.asarray([1, 1, 1])
+        b = xp.asarray([2, 2, 2])
```

Comment:
I guess it is a similar point here about the original test using an array-like, so it might be possible to interpret this as a slightly different regression test.

Reply:
Yeah, if we want to be thorough then I would suggest adding a new test to explicitly check for the array-like to numpy-array behaviour. These changes overall are good though - Ralf mentioned on a different PR that clearly the vast majority of tests which use array-like input instead of NumPy arrays do so out of brevity rather than a desire to test that (and the same goes for dtypes as the default dtypes differ between backends, while many tests are written without a dtype specified).

Reply:
The parameter is document to accept an array-like. I'll add a separate test for array-like input for `np_only`.

Reply:
addressed in [cc58124](https://github.com/scipy/scipy/pull/19682/commits/cc581243827c226caebdbbd8041ad08a224b1006)